### ISSUE TITLE
How to deal with previous commits within the CI?

### ISSUE THREAD

**Issue by dudicoco** [yuvipanda/pre-commit-hook-ensure-sops#8]
Hi,

I'm trying to figure out how to solve this issue when running the hook via `pre-commit` within a CI system:
1. User commits an unencrypted secret and opens a PR
2. `pre-commit` runs on the commit within the CI and the hook fails due to unencrypted secret
3. User encrypts the secret and pushes a new commit to the remote branch
4. `pre-commit` exists with success within the CI
5. The old commit with the unencrypted secret still exists within the branch and within the PR

Basically, it is required to somehow runs the hook on each of the commits in the branch in order to solve this, the user will have to force push a commit and overwrite the previous commit with the unencrypted secret.

**Comment by mentos1386** [comment]
@dudicoco i think this is supposed to be run on the "client side". Before the 1. Steps even happens.

Basically when you do `git commit` this would be run and fail if unencrypted secret exists. Thus, the PR couldn't even be created in first place.

**Comment by dudicoco** [comment]
@mentos1386 of course it will also run on the client, but it should also run on your CI.

You can't assume a user ran pre-commit locally and fixed all of the issues, therefor pre-commit should always run on the CI as well for validation.
See example: https://github.com/pre-commit/action
